services/workflows-service/src/customer/customer.controller.external.ts

```diff
@@ -37,8 +38,16 @@ export class CustomerControllerExternal {
       };
     }
 
+    const apiKey = customer.authenticationConfiguration?.authValue ?? randomUUID();
```

Comment:
Isn't a falsy `authValue` something we should throw an error on?

Reply:
No, the ideal is that the client don't provide the API key, but the server will generate a random one and return it.